brushfire-training/src/main/scala/com/stripe/brushfire/training/Evaluators.scala

```diff
@@ -0,0 +1,49 @@
+package com.stripe.brushfire.training
+
+import com.stripe.brushfire._
+import com.twitter.algebird._
+
+case class ChiSquaredEvaluator[L, W](implicit weightMonoid: Monoid[W], weightDouble: W => Double)
+    extends Evaluator[Map[L, W]] {
+  def trainingError(leaves: Iterable[Map[L,W]]) = {
+    val rows = leaves.filter { _.nonEmpty }
+    if (rows.size > 1) {
+      val n = weightMonoid.sum(rows.flatMap { _.values })
+      val rowTotals = rows.map { row => weightMonoid.sum(row.values) }.toList
+      val columnKeys = rows.flatMap { _.keys }.toList
```

Comment:
Hrmm, no, I think we do want a set, since we get our per-row iteration with the rows.zipWithIndex.